G10: Post `docker pull` commands for local testing

dsotirakis opened the pull request:
**What is this feature?**

Makes it able for devs to test their changes per commit, using a docker image. The docker image will appear as a check run after the `publish-images` step has successfully ran.

**Why do we need this feature?**

As part of the testing phase before Grafana 10

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes https://github.com/grafana/grafana-delivery/issues/121

dsotirakis commented:
Use the following command to pull the docker image:

`docker pull grafana/grafana:test`

dsotirakis committed:
Lint starlark

Add forgotten secrets

Lint starlark

Change secrets

zerok reviewed (changes requested):
We might need a way to restrict building these images further so that only specific PRs would do that.

dsotirakis committed:
Add grabpl publish images

Update grabpl - add github pr token

dsotirakis committed:
Replace github_pr_token with gh app

# Conflicts:
#	.drone.yml
#	scripts/drone/vault.star

# Conflicts:
#	.drone.yml

# Conflicts:
#	.drone.yml

dsotirakis committed:
Update grabpl to v3.0.35

dsotirakis commented:
> We might need a way to restrict building these images further so that only specific PRs would do that.

@zerok discussed this already - we will use a GitHub app which restricts that to org members only :)

dsotirakis committed:
Lint starlark

zerok commented on scripts/drone/vault.star:
```diff
@@ -123,6 +123,16 @@ def secrets():
+            "infra/data/ci/grafana-release-eng/grafana-enterprise",
+            "upstream-repo-main",
+        ),
+        vault_secret(
```
This one doesn't seem to be used anywhere :)

dsotirakis replied:
Oops, massive c/p :D Thanks!

dsotirakis committed:
Remove unnecessary secrets
